File: Ex4/SymbolReplacer.cpp
#include <iostream>

std::string *
ReplaceSymbols(std::string &String)
{
    std::string *NewString = new std::string;

    for(char c : String) {
        switch(c) {
            case '&':
                *NewString += "&amp;";
                break;
            case '<':
                *NewString += "&lt;";
                break;
            case '>':
                *NewString += "&gt;";
                break;
            default:
                *NewString += c;
                break;
        }
    }

    return NewString;
}

int
main(int ArgCount, char **ArgV)
{
    std::string InputString(ArgV[1]);
    std::cout << *ReplaceSymbols(InputString) << std::endl;
}

File: Ex4/SymbolReplacer.rs
use std::env;

fn replace_symbols(string: String) -> String {
    let mut new_string = String::new();

    for c in string.chars() {
        match c {
            '&' => new_string.push_str("&amp;"),
            '>' => new_string.push_str("&gt;"),
            '<' => new_string.push_str("&lt;"),
            _ => new_string.push(c),
        }
    }

    return new_string
}

fn main() {
    let args: Vec<String> = env::args().collect();
    let new_string = replace_symbols(args.get(1).unwrap().to_string());
    println!("{}", new_string);
}
